src/ngx_http_lua_socket_tcp.c

```diff
@@ -4656,67 +5216,31 @@ ngx_http_lua_socket_tcp_setkeepalive(lua_State *L)
             pool_size = llcf->pool_size;
         }
 
-        if (pool_size == 0) {
-            lua_pushnil(L);
-            lua_pushliteral(L, "zero pool size");
-            return 2;
-        }
-
-        ngx_log_debug1(NGX_LOG_DEBUG_HTTP, r->connection->log, 0,
-                       "lua tcp socket connection pool size: %ui", pool_size);
-
-        key_len = ngx_align(key.len + 1, sizeof(void *));
-
-        size = sizeof(ngx_http_lua_socket_pool_t) + key_len - 1
-               + sizeof(ngx_http_lua_socket_pool_item_t)
-               * pool_size;
-
-        spool = lua_newuserdata(L, size);
-        if (spool == NULL) {
-            return luaL_error(L, "no memory");
+        if (pool_size <= 0) {
```

Comment:
I think we should issue a warning log message saying `argument "pool_size" ignored; use new() method instead`.